app/controllers/posts_controller.rb

```diff
@@ -6,4 +6,65 @@ def index
   def show
     @post = Post.where(user_id: params[:user_id]).find(params[:id])
   end
+
+  def new
+    post = Post.new
+    respond_to do |format|
+      format.html { render :new, locals: { post: } }
+    end
+  end
+
+  def new_comment
+    comment = Comment.new
+    respond_to do |format|
+      format.html { render :new_comment, locals: { comment: } }
+    end
+  end
+
+  def create_comment
+    comment = Comment.new(params.require(:comment).permit(:text))
+    comment.user = current_user
+    comment.post = Post.find(params[:id])
+
+    respond_to do |format|
+      format.html do
+        if comment.save
+          flash[:success] = 'Comment created successfully'
+          redirect_to '/users/1/posts/1/'
+        else
+          flash[:error] = 'Comment faild to create'
+          render :new_comment, locals: { comment: }
+        end
+      end
+    end
+  end
```

Comment:
- [ ] Kindly move these actions to a new controller. There should be a **CommentsController** just as we have **PostsController**. Kindly remember to rename the actions to their appropriate names. `new_comment` should be just `new`; and `create_comment` should just be `create` in the new controller.

Reply:
Yeah, I thought it was a  good idea to have the comments and likes controllers handle the element's behavior on their respective controllers. I just didn't commit to the thought process 😄

Reply:
This change has been done well